src/libPMacc/include/particles/ParticlesBase.kernel

```diff
@@ -494,7 +495,7 @@ __global__ void kernelBashParticles(ParticlesBox<FRAME, Mapping::Dim> pb,
 
     DataSpace<Dim> superCellIdx = mapper.getSuperCellIndex(DataSpace<Dim > (blockIdx));
 
-    __shared__ uint32_t numBashedParticles;
+    __shared__ int numBashedParticles;
```

Comment:
`nvidia::atomicAllInc` is using `int` to communicate values between threads inside of a warp. This might be half the largest value to `2^31-1` (for a `unsigned int` cast to `int`).
Currently the most clean way is that only int is supported so that the user must cast there values.